TLM/TLM/LoadingExtension.cs

```diff
@@ -169,86 +172,74 @@ public class LoadingExtension : LoadingExtensionBase {
-                                    "version yet and thus it is very likely it will not work as expected.",
-                                    BuildConfig.applicationVersion,
-                                    BuildConfig.VersionToString(TrafficManagerMod.GAME_VERSION, false));
-
-                                Log.Error(msg);
-                                Singleton<SimulationManager>.instance.m_ThreadingWrapper.QueueMainThread(
-                                        () => {
-                                            UIView.library
-                                                  .ShowModal<ExceptionPanel>("ExceptionPanel")
-                                                  .SetMessage(
-                                                      "TM:PE has not been updated yet",
-                                                      msg,
-                                                      false);
-                                        });
-                            } else if (isModNewer) {
-                                string msg = string.Format(
-                                    "Traffic Manager: President Edition has been built for game version {0}. " +
-                                    "You are running game version {1}. Some features of TM:PE will not " +
-                                    "work with older game versions. Please let Steam update your game.",
-                                    BuildConfig.VersionToString(TrafficManagerMod.GAME_VERSION, false),
-                                    BuildConfig.applicationVersion);
-
-                                Log.Error(msg);
-                                Singleton<SimulationManager>
-                                    .instance.m_ThreadingWrapper.QueueMainThread(
-                                        () => {
-                                            UIView.library
-                                                  .ShowModal<ExceptionPanel>("ExceptionPanel")
-                                                  .SetMessage(
-                                                      "Your game should be updated",
-                                                      msg,
-                                                      false);
-                                        });
-                            }
-                        }
-
-                        IsGameLoaded = true;
-                        break;
-                    }
-
-                default: {
-                        Log.Info($"OnLevelLoaded: Unsupported game mode {mode}");
-                        return;
-                    }
+            if (BuildConfig.applicationVersion != BuildConfig.VersionToString(
+                    TrafficManagerMod.GAME_VERSION,
+                    false)) {
+                string[] majorVersionElms = BuildConfig.applicationVersion.Split('-');
+                string[] versionElms = majorVersionElms[0].Split('.');
```

Comment:
would kindly raise an improvement idea:
I would use a regex pattern ( for instance: \d+.\d+-\d+ ) to parse the version elements.
In my view, regex parsing could avoid chain calling such as did in 179 line and would also be more readable. 

More about Regex in C#
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.match?view=netcore-3.1

Reply:
again ignore white spaces.